inference/core/workflows/core_steps/transformations/size_measurement/v1.py

```diff
@@ -0,0 +1,171 @@
+from typing import List, Literal, Optional, Tuple, Type, Union
+
+import cv2 as cv
+import numpy as np
+import supervision as sv
+from pydantic import ConfigDict, Field, validator
+
+from inference.core.workflows.execution_engine.entities.base import (
+    Batch,
+    OutputDefinition,
+)
+from inference.core.workflows.execution_engine.entities.types import (
+    INSTANCE_SEGMENTATION_PREDICTION_KIND,
+    LIST_OF_VALUES_KIND,
+    OBJECT_DETECTION_PREDICTION_KIND,
+    StepOutputSelector,
+)
+from inference.core.workflows.prototypes.block import (
+    BlockResult,
+    WorkflowBlock,
+    WorkflowBlockManifest,
+)
+
+OUTPUT_KEY: str = "dimensions"
+TYPE: str = "roboflow_core/size_measurement@v1"
+SHORT_DESCRIPTION = (
+    "Measure the dimensions of objects in relation to a reference object."
+)
+LONG_DESCRIPTION = """
+The `SizeMeasurementBlock` is a transformer block designed to measure the dimensions of objects
```

Comment:
The way on how the block works is hard to be understood from description - lack of mention that `reference_predictions` may provide multiple boxes, but the most confident one will be selected - we shall also suggest to combine the block with "filter + potentially continue if" when different behaviour is expected.